nautilus-extension/nautilus-gsconnect.py

```diff
@@ -2,85 +2,75 @@
 nautilus-gsconnect.py - A Nautilus extension for sending files via GSConnect.
 
 A great deal of credit and appreciation is owed to the indicator-kdeconnect
-developers for the sister Python script 'kdeconnect-send-nautilus.py':
+developers for the sister Python script "kdeconnect-send-nautilus.py":
 
 https://github.com/Bajoja/indicator-kdeconnect/blob/master/data/extensions/kdeconnect-send-nautilus.py
 """
 
-import gettext
-import os.path
-import sys
+from gettext import translation
+from os.path import join, exists
+from sys import argv
 
 import gi
-gi.require_version('Gio', '2.0')
-gi.require_version('GLib', '2.0')
-gi.require_version('GObject', '2.0')
-from gi.repository import Gio, GLib, GObject
+gi.require_version("Gio", "2.0")
+gi.require_version("GLib", "2.0")
+gi.require_version("GObject", "2.0")
+from gi.repository.Gio import (
+    DBusProxy, DBusProxyFlags, BusType, DBusActionGroup, DBusCallFlags)
+from gi.repository.GLib import get_user_data_dir, Variant
+from gi.repository.GObject import Object
 
 
 # Host application detection
 #
-# Nemo seems to reliably identify itself as 'nemo' in argv[0], so we
+# Nemo seems to reliably identify itself as "nemo" in argv[0], so we
 # can test for that. Nautilus detection is less reliable, so don't try.
 # See https://github.com/linuxmint/nemo-extensions/issues/330
-if "nemo" in sys.argv[0].lower():
+if "nemo" in argv[0].lower():
     # Host runtime is nemo-python
-    gi.require_version('Nemo', '3.0')
+    gi.require_version("Nemo", "3.0")
     from gi.repository import Nemo as FileManager
 else:
     # Otherwise, just assume it's nautilus-python
     from gi.repository import Nautilus as FileManager
 
 
-SERVICE_NAME = 'org.gnome.Shell.Extensions.GSConnect'
-SERVICE_PATH = '/org/gnome/Shell/Extensions/GSConnect'
+SERVICE_NAME = "org.gnome.Shell.Extensions.GSConnect"
+SERVICE_PATH = "/org/gnome/Shell/Extensions/GSConnect"
 
 # Init gettext translations
-LOCALE_DIR = os.path.join(GLib.get_user_data_dir(),
-                          'gnome-shell', 'extensions',
-                          'gsconnect@andyholmes.github.io', 'locale')
+LOCALE_DIR = join(
```

Comment:
Here again: `os.path.join` is immediately clear to future readers of this code. But reading your version, what's the definition of `join()`? Is it a local function? Is it `str.join()`? `Thread.join()`? Importing generically-named functions from specific packages, when there are at least a half-dozen possible modules it could have come from, hurts code readability far more than the extra characters required for parent module chains.

Reply:
> To be fair some people instead import it as joinpath.

At which point, you're saving a whopping 4 characters over `os.path.join`, and you have to remember this random other name for a standard library function. Where's the _benefit_?

> It doesn't get confused with str.join, because one is clearly a method while the other a function. 

That's a fair point that also covers `Thread.join()`, granted.

> And 9/10 times the variables past to join have the variable name "path" so yeah

True, but then you have to _analyze_ the entire line. The [human] parsing goes from, "Ok, I'm joining a path, and the components are..." (completely linear) to "OK, I'm joining something... these looks like path components... that must've been `os.path.join` back there..." (requires backtracking).

Even if it's not "unclear", it can be "clear[-ish]" while still being "less obvious", and that can still be a negative. If interpreting the less-obvious line might require an extra second or two of thought, once you multiply that by the number of times this code will be read by someone over its lifetime you're talking real numbers.